packages/react-native/ReactCommon/jsc/JSCRuntime.cpp

```diff
@@ -398,13 +412,6 @@ JSCRuntime::JSCRuntime(JSGlobalContextRef ctx)
       stringCounter_(0)
 #endif
 {
-#ifndef NDEBUG
-#ifdef TARGET_OS_MAC
-  if (__builtin_available(macOS 13.3, iOS 16.4, tvOS 16.4, *)) {
```

Comment:
This code seems to be specific to Apple platforms.
It is better to add another check for Clang compiler and/or target platform. This code is in ReactCommon, it will fail to compile in MSVC for Windows if someone wants to do it.

Reply:
Current iteration takes my approach from https://github.com/facebook/react-native/pull/39037 and puts it behind an `#ifdef _JSC_HAS_INSPECTABLE` check, which right now is apple specific. That should be fine?